.github/workflows/set-vars.yml

```
@@ -0,0 +1,24 @@
+on:
+    workflow_call:
+        outputs:
+            CI_IMAGE:
+                value: ${{ jobs.set_vars.outputs.CI_IMAGE }}
+            CI_REGISTRY:
+                value: ${{ jobs.set_vars.outputs.CI_REGISTRY }}
+
+env:
+    REGISTRY: ghcr.io
+
+jobs:
+
+    set_vars:
+        runs-on: [ubuntu-latest]
+        outputs:
+            CI_IMAGE   : ${{ steps.common.outputs.CI_IMAGE }}
+            CI_REGISTRY: ${{ steps.common.outputs.CI_REGISTRY }}
+        steps:
+            - name: Export common variables.
+              id  : common
+              run : |
+                  echo "CI_IMAGE=${{ env.REGISTRY }}/${{ github.repository }}/hwloc_xml_parser" >> $GITHUB_OUTPUT
```

Let's keep the image name aligned with the repo name.
```suggestion
                  echo "CI_IMAGE   =${{ env.REGISTRY }}/${{ github.repository }}:latest" >> $GITHUB_OUTPUT
```

For now your image name looks like `echo "CI_IMAGE=ghcr.io/uliegecsm/hwloc-xml-parser/hwloc_xml_parser" >> $GITHUB_OUTPUT` which does not seem right.